### PR TITLE
DOC:add append example in to_excel documentation

### LINKED ISSUE #27051: Improve to_excel Append Documentation
#### Code Sample, a copy-pastable example if possible
I have recently worked on a project where I needed to insert pandas dataframes into an existing Excel workbook (.xlsx) file.

I wrote the following function:
 
```python

def append_df_to_excel(filename, df, sheet_name='Sheet1', startcol='A', startrow=1, **to_excel_kwargs):
    """
    Append a DataFrame [df] to existing Excel file [filename]
    into [sheet_name] Sheet.
    If [filename] doesn't exist, then this function will create it.

    Parameters:
      filename : File path or existing ExcelWriter
                 (Example: '/path/to/file.xlsx')
      df : dataframe to save to workbook
      sheet_name : Name of sheet which will contain DataFrame.
                   (default: 'Sheet1')
      startrow : upper left cell row to dump data frame - Excel row (starting from 1) not Python row (starting from 0)
      startcol : upper left cell column to dump data frame - Excel column (character)
      to_excel_kwargs : arguments which will be passed to `DataFrame.to_excel()`
                        [can be dictionary]

    Returns: None
    """

    # ignore [engine] parameter if it was passed
    if 'engine' in to_excel_kwargs:
        to_excel_kwargs.pop('engine')

    writer = pd.ExcelWriter(filename, engine='openpyxl')

    try:
        # try to open an existing workbook
        writer.book = load_workbook(filename)

        # copy existing sheets
        writer.sheets = {ws.title:ws for ws in writer.book.worksheets}

    except FileNotFoundError:
        # file does not exist yet, we will create it
        pass

    # Convert Excel rows into Python rows
    if startrow > 0:
        startrow = startrow - 1

    # Convert Excel columns into Python columns
    startcol = col2num(startcol) - 1

    # write out the new sheet
    if 'header' in to_excel_kwargs:
        header = to_excel_kwargs['header']
        del(to_excel_kwargs['header'])
    else:
        header = False

    df.to_excel(writer, sheet_name, startrow=startrow, startcol=startcol, header=header, index=False, **to_excel_kwargs)

    writer.save()

```

Could this be optional behaviour of df.to_excel()

#### Problem description

It is sometimes necessary to fill data into templates provided in excel without overwriting other data or formulae in the excel workbook. Being able to do so using df.to_excel (and then developing the functions to make this relatively efficient if you have to write to many places in the same file) seems a natural and useful extension.

### PR DESCRIPTION
**Pull request by sameshl**
closes #27051

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -2184,6 +2184,12 @@ def _repr_data_resource_(self):
     automatically chosen depending on the file extension):
 
     >>> df1.to_excel('output1.xlsx', engine='xlsxwriter')  # doctest: +SKIP
+
+    To append data to an existing excel file:
+
+    >>> with pd.ExcelWriter('the_existing_file.xlsx', engine='openpyxl',
+    ...                     mode='a') as writer:
+    ...     df.to_excel(writer)
```
Comment (datapythonista) [inline pandas/core/generic.py]: This example feels disconnected with the rest of the docstring.

I'd move it to line 2181, and say something like `ExcelWriter can also be used to append to an existing Excel file`. And then use the same file `output.xlsx` and append a sheet `Sheet_name_3`. I think that will be much clearer.

You also need the `# doctest: +SKIP` as in the other examples.

Comment (sameshl) [reply pandas/core/generic.py]: Yeah. You are right. That will be clearer. Will make the required change

**Review by datapythonista — APPROVED** [review]
lgtm, thanks for taking care of this @sameshl

**Comment by TomAugspurger** [comment]
Merged master to hopefully fix CI. Ping on green.

**Comment by sameshl** [comment]
Will do @TomAugspurger

**Comment by sameshl** [comment]
@TomAugspurger Looks good now!
Btw, could you explain why the build was failing and merging master to my branch fixed it? :sweat_smile: 
Thanks!

**Comment by datapythonista** [comment]
Thanks @sameshl 

If there is something broken in master (e.g. a pandas dependency has been updated and it causes a test to fail), then all PRs will fail even if they are correct. So, master needs to be fixed first, and then the fixes in master need to be merged into the PRs that failed.

**Comment by sameshl** [comment]
@datapythonista Thanks for the explanation!

**Pull request merged**